test/PrimeNetwork.t.sol

```diff
@@ -937,4 +938,63 @@ contract PrimeNetworkTest is Test {
         vm.expectRevert(bytes("ComputePool: pool is at capacity"));
         nodeJoin(domain, pool, provider_good1, node_good2);
     }
+
+    function test_RewardFunction() public {
+        uint256 domain = newDomain("Decentralized Training", "https://primeintellect.ai/training/params");
+        uint256 pool = newPool(domain, "INTELLECT-1", "https://primeintellect.ai/pools/intellect-1");
+
+        startPool(pool);
+
+        addProviderWithStake(provider_good1, minStake * 100);
+        whitelistProvider(provider_good1);
+
+        addNodeWithCompute(provider_good1, node_good1, node_good1_sk, computeUnitsPerNode);
+
+        validateNode(provider_good1, node_good1);
+
+        assertEq(computeRegistry.getNode(provider_good1, node_good1).subkey, node_good1);
+
+        nodeJoin(domain, pool, provider_good1, node_good1);
+
+        assertEq(isNodeInPool(pool, node_good1), true);
+
+        IRewardsDistributor rewardsDistributor = IRewardsDistributor(computePool.getRewardDistributorForPool(pool));
```

Comment:
wonder if it's worth testing the rewards claiming on all the different RewardsDistributor types since they all have different logic to calculate rewards? as a sanity check.

Reply:
it's hard to do here because the rewards factory is set when the compute pool contract is deployed (rather than when a pool is created). all the different claiming flows are tested in their individual test contracts, but to test them all here would require a lot of changes to how the factory setup works. we should def simplify/rework that too later but i think better as a separate task/PR.

Reply:
makes sense!